pkg/enforcement/check.go

```diff
@@ -0,0 +1,99 @@
+package enforcement
+
+import (
+	"bytes"
+	"encoding/json"
+	"github.com/permitio/permit-golang/pkg/errors"
+	"go.uber.org/zap"
+	"net/http"
+	"time"
+)
+
+type Action string
+
+type CheckRequest struct {
+	User     User              `json:"user"`
+	Action   Action            `json:"action"`
+	Resource Resource          `json:"resource"`
+	Context  map[string]string `json:"context"`
+}
+type CheckResponse struct {
+	allow string
+}
+
+func NewCheckRequest(user User, action Action, resource Resource, context map[string]string) *CheckRequest {
+	return &CheckRequest{
+		User:     user,
+		Action:   action,
+		Resource: resource,
+		Context:  context,
+	}
+}
+
+func (e *PermitEnforcer) Check(user User, action Action, resource Resource, context ...map[string]string) (bool, error) {
+	const (
+		reqMethod           = "POST"
+		reqContentTypeKey   = "Content-Type"
+		reqContentTypeValue = "application/json"
+		reqAuthKey          = "Authorization"
+	)
+	var checkRes map[string]interface{}
+	reqAuthValue := "Bearer " + e.config.GetToken()
+	reqEndpoint := e.config.GetPdpUrl() + "/allowed"
+
+	if context == nil {
```

Comment:
rename to `permitContext` / `scopeContext` / whatever. 
(This is the last comment I'll make regarding that - just make sure we're not using context alone so we won't confuse with glang's context)